File: core/src/main/java/moe/caa/multilogin/core/auth/validate/ValidateContext.java
package moe.caa.multilogin.core.auth.validate;

import lombok.Data;
import moe.caa.multilogin.api.profile.GameProfile;
import moe.caa.multilogin.core.auth.service.BaseServiceAuthenticationResult;

/**
 * 游戏内验证消息上下文
 */
@Data
public class ValidateContext {
    private final BaseServiceAuthenticationResult baseServiceAuthenticationResult;

    private final GameProfile inGameProfile;
    private String disallowMessage;
    private boolean needWait;
    private boolean onlineNameUpdated = false;


    protected ValidateContext(BaseServiceAuthenticationResult baseServiceAuthenticationResult) {
        this.baseServiceAuthenticationResult = baseServiceAuthenticationResult;
        this.inGameProfile = baseServiceAuthenticationResult.getResponse().clone();
    }
}

File: velocity/src/main/kotlin/moe/caa/multilogin/velocity/auth/validate/ValidateContext.kt
package moe.caa.multilogin.velocity.auth.validate

import moe.caa.multilogin.api.profile.GameProfile
import moe.caa.multilogin.velocity.auth.OnlineGameData
import moe.caa.multilogin.velocity.config.service.BaseService

data class ValidateContext(
    val baseService: BaseService,
    val userGameProfile: GameProfile
) {
    lateinit var profileGameProfile: GameProfile

    fun toGameData() = OnlineGameData(
        userGameProfile,
        baseService,
        profileGameProfile
    )
}
